proposals/XXXX-unicode-normalization.md

```diff
@@ -816,6 +864,20 @@ for scalar in nfcString.unicodeScalars.normalized.nfc {
 
 `nfcString` may have an internal performance bit set so we know it is NFC. If it does, the streaming normaliser in the `for`-loop detects this and knows it doesn't need to normalise the contents a second time. It would be nice if we could generalise this, or at least take note of a normalised prefix that we can skip.
 
+### Normalizing intializers
+
+We could define initializers similar to `String(decoding:as)`, e.g. `String(normalizing:)`, which decode and transform the contents into the stdlib's preferred normal form without always needing to create an intermediary `String` instance. Such strings will compare much faster to each other, but may have different contents when viewed under the `UnicodeScalarView`, `UTF8View`, or `UTF16View`.
+
+### Normalized code units
+
+Normalization API is presented in terms of `Unicode.Scalar`s, but a lower-level interface might run directly on top of (validated) UTF-8 code units. Many useful normalization fast-paths can be implemented directly on top of UTF-8 without needing to decode. 
+
+For example, any `Unicode.Scalar` that is less than `U+0300` is in NFC and can be skipped over for NFC normalization. In UTF-8, this amounts to skipping until we find a byte that is `0xCC` or larger.
+
```

Comment:
My thought is that rather than have `UTF8Span` vend a stateful closure, we might want to have `UTF8Span` be a basis of low-level fast normal views. But that might be pending non-escapable sequences/iterators.